### ISSUE TITLE
Error "Cannot find type Microsoft.Toolkit.Uwp.UI.Behaviors.BehaviorBase`1 in module Microsoft.Toolkit.Uwp.UI.dll"

### ISSUE THREAD

**Issue by zipswich** [windows-toolkit/Graph-Controls#102]
## Describe the bug
The app has been using InteractiveProviderBehavior for a very long time.  Toolkit NuGet packages were updated to V7.0.1, and Microsoft.Toolkit.Graph.Controls was updated to 7.0.0-preview2.  Now, I am getting the following compilation error:
"Xaml Internal Error error WMC9999: Cannot find type Microsoft.Toolkit.Uwp.UI.Behaviors.BehaviorBase`1 in module Microsoft.Toolkit.Uwp.UI.dll."

- [x] Is this bug a regression in the toolkit?  If so, what toolkit version did you last see it work: 6.1.0-preview1

## Steps to Reproduce
Steps to reproduce the behavior (the code is solely for debugging":
1. Copy the following code form the Toolkit sample app to an Xaml page:
```
    <Interactivity:Interaction.Behaviors>
        <providers:InteractiveProviderBehavior ClientId="YOUR_CLIENT_ID_HERE" Scopes="User.Read,User.ReadBasic.All,People.Read,Calendars.Read"/>
    </Interactivity:Interaction.Behaviors>
```

## Expected behavior
The app will compile.

## Screenshots
![image](https://user-images.githubusercontent.com/6095671/117165730-c1506000-ad93-11eb-9e95-536c48af8ef8.png)

## Environment

```
NuGet Package(s): 

![image](https://user-images.githubusercontent.com/6095671/117165975-f6f54900-ad93-11eb-9ea4-d54e21b3acc1.png)

Windows 10 Build Number:
- [ ] Fall Creators Update (16299)
- [ ] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [x] May 2019 Update (18362)
- [ ] Insider Build (build number: )

App min and target version:
- [ ] Fall Creators Update (16299)
- [ ] April 2018 Update (17134)
- [ ] October 2018 Update (17763)
- [x] May 2019 Update (18362)
- [ ] Insider Build (xxxxx)

Device form factor:
- [x] Desktop
- [ ] Xbox
- [ ] Surface Hub
- [ ] IoT

Visual Studio 
- [ ] 2017 (version: )
- [x] 2019 (version: 16.9.4) 
- [ ] 2019 Preview (version: )

```

**Comment by shweaver-MSFT** [comment]
Hi @zipswich, we've made some major changes to the expected usage pattern for the Graph-Controls authentication providers. We no longer support using behaviors to configure the GlobalProvider from XAML, and instead prefer for you to integrate from your App.xaml.cs, or similar startup location.

I apologize for any inconvenience though, I've been making some heavy changes recently so that we can get our initial release and haven't communicated the plan outwardly very well. Here is a link to the [Release Plan](https://github.com/windows-toolkit/Graph-Controls/issues/81)

You can see the new pattern described in the [Quickstart Guide for the new WindowsProvider](https://github.com/windows-toolkit/Graph-Controls/blob/main/Docs/QuickStart-WindowsProvider.md).

A similar pattern will work for the MsalProvider as well. Check it out and let us know if you have any feedback!

**Comment by michael-hawker** [comment]
There's more info about using our [Preview Feeds in our Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki/Preview-Packages). FYI the name of this package will probably be updated soon.

**Comment by zipswich** [comment]
Thank you.  I have installed the package.
I cannot figure out why I am still getting:

> C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\XamlCompiler\Microsoft.Windows.UI.Xaml.Common.targets(410,5): Xaml Internal Error error WMC9999: Cannot find type Microsoft.Toolkit.Uwp.UI.Behaviors.BehaviorBase`1 in module Microsoft.Toolkit.Uwp.UI.dll.

I did a global search of "Behaviors" and could not find a single instance.  I cleaned the project, deleted obj and bin folders and restarted VS to no avail.

Could anyone offer a tip on where I should look for the culprit?

**Comment by michael-hawker** [comment]
@zipswich can you share the new list of package/versions you have installed in the project?

`BehaviorBase` was moved in 7.0.0, but as long as you're no longer mixing the current Toolkit packages with the old Graph packages, then you should be fine, as the latest Graph packages from the dev feed don't contain a behaviors reference anymore.

**Comment by shweaver-MSFT** [comment]
@zipswich, I think the `Microsoft.Toolkit.Graph` and `Microsoft.Toolkit.Graph.Controls` packages are the problem. The new equivalents would be: `CommunityToolkit.Net.Graph` and `CommunityToolkit.Uwp.Controls`. Also try to see if you can get the versions synced up between the `CommunityToolkit.*` packages.

**Comment by shweaver-MSFT** [comment]
You got it! My mistake 😊 `CommunityToolkit.Uwp.Graph.Controls` is the correct package.

**Comment by zipswich** [comment]
Thank you.  
I use the kit to handle OneDrive.  The typical code is like the following:
```
DriveItem driveItem = new DriveItem
{
    Name = "My App",
    Folder = new Folder
    {
    },
    AdditionalData = new Dictionary<string, object>()
        {
            {"@microsoft.graph.conflictBehavior","fail"}
        }
};

try
{
    await provider.Graph.Me.Drive.Root.Children.Request().AddAsync(driveItem);
}
```
Is there a new guideline about how to adapt such code for the new packages? IProvider.Graph no longer exists.

**Comment by shweaver-MSFT** [comment]
Haha, I really need to write some docs for this 😅 I'll start on them this week so you can have a better picture of how to use the new pattern.

I did a little shuffling across the packages to decouple any of the Graph SDK usage from the authentication providers. This is helpful for some of our partners who don't want the whole Graph SDK, and prefer to make the calls manually.

So the new way to access the GraphServiceClient instance is through an extension method on IProvider. The extension lives in a separate package, `CommunityToolkit.Net.Graph`. However, the controls package depends on it already, so you shouldn't need to install it explicitly.  

Here is a link to the source so you can see how it works: https://github.com/windows-toolkit/Graph-Controls/blob/main/CommunityToolkit.Net.Graph/Extensions/ProviderExtensions.cs

```
using CommunityToolkit.Net.Authentication;
using CommunityToolkit.Net.Graph.Extensions;

var provider = ProviderManager.Instance.GlobalProvider;
var graphClient = provider.GetClient();

var me = await graphClient.Me.Request().GetAsync();
```

**Comment by zipswich** [comment]
@shweaver-MSFT  Thank you.  It looks great so far.
I am wondering if you could give me a quick tip.   The app needs to create its own folder on OneDrive, then write and read in that folder.  I am uisng `Scopes="Files.ReadWrite.All"`.  I suspect that is too broad.  What scope should I use?
![image](https://user-images.githubusercontent.com/6095671/118041308-a015e380-b340-11eb-9769-c949c5b77a71.png)

**Comment by zipswich** [comment]
@shweaver-MSFT I think `Files.ReadWrite.AppFolder` may be the best, but I do not know how to get with ` graphClient.Me.Drive`.

**Comment by shweaver-MSFT** [comment]
I agree, `Files.ReadWrite.AppFolder` sounds perfect. I do some OneDrive file work in the special app folder in the [OneDriveDataSource](https://github.com/windows-toolkit/Graph-Controls/blob/main/CommunityToolkit.Uwp.Graph.Controls/Helpers/RoamingSettings/OneDriveDataSource.cs) (part of a new helper for roaming settings that is in preview)

Here is an example:
```
public static async Task<T> Retrieve<T>(string userId, string fileWithExt, IObjectSerializer serializer)
{
    Stream stream = await Graph.Users[userId].Drive.Special.AppRoot.ItemWithPath(fileWithExt).Content.Request().GetAsync();

    string streamContents = new StreamReader(stream).ReadToEnd();

    return serializer.Deserialize<T>(streamContents);
}
```

Hope this helps!

**Comment by zipswich** [comment]
@shweaver-MSFT Thanks a lot.  The following code works PERFECTLY:
```
driveItemFile = await graphClient.Me.Drive.Special.AppRoot
    .ItemWithPath(sf.Name)
    .Content
    .Request()
    .PutAsync<DriveItem>(localStream);
```
I should be all set now.   
I think the migration to new packages is easy and straightforward as long as one knows the instructions of the few simple steps.

**Comment by michael-hawker** [comment]
Thanks for all the help and feedback @zipswich! Glad we've got you back up and working. @shweaver-MSFT's working on new docs for all these new packages and experiences, so hopefully we'll have this all live in a package on NuGet in a few weeks. That's our hope at least! 🤞

You're early vetting has been invaluable! 🦙❤

We'll keep you posted on if anything significant changes, as I mentioned the package names are going to get jumbled around a tiny bit in the next week or so, but beyond that I don't know if we see any other big shuffles at the moment yet. If so it may just be moving code from one package to another to make things more accessible to .NET vs. UWP.

**Comment by zipswich** [comment]
@michael-hawker Thank you for all the effort.  Actually, [you walked me through the previous code over a year ago](https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3097). 
It would be great if you could drop a note here when the big release finally comes. 
I do not use LoginButton.  Instead, I use `await provider.LoginAsync();`. It works great. The UI is perfect.

**Comment by johngagefaulkner** [comment]
I know it has been 26 days but I thought I was about to spend **26 hours** trying to figure out this issue. 
I mean, _when was the last time you saw the 2nd page of your Google search results?_

Thank you guys so much!
